components/AdventurerStatus.ts

```diff
@@ -215,4 +222,92 @@ export class AdventurerStatus extends HTMLElement {
   }
 }
 
+import { getFlowState } from '../game/utils';
+
 customElements.define('adventurer-status', AdventurerStatus);
+
+export class FlowChart extends HTMLElement {
+  private _skill: number = 50;
+  private _challenge: number = 50;
+  private _canvas: HTMLCanvasElement | null = null;
+  private _ctx: CanvasRenderingContext2D | null = null;
+
+  static get observedAttributes() {
+    return ['skill', 'challenge'];
+  }
+
+  constructor() {
+    super();
+  }
+
+  connectedCallback() {
+    this.innerHTML = `
+      <div class="relative" style="width: 100px; height: 100px;">
+        <canvas width="100" height="100"></canvas>
+        <p class="absolute bottom-0 left-0 right-0 text-center text-xs pointer-events-none">Skill</p>
+        <p class="absolute top-0 left-0 bottom-0 text-center text-xs pointer-events-none" style="writing-mode: vertical-rl; transform: rotate(180deg);">Challenge</p>
```

Comment:
these should use font-family "Pixelated MS Sans Serif" with a 1px white outline. Feel free to do it in canvas rather than css if necessary.